test/BasicAccessFactoryTest.php

```diff
@@ -15,45 +15,60 @@
 use PHPUnit\Framework\Assert;
 use PHPUnit\Framework\TestCase;
 use Prophecy\Prophecy\ObjectProphecy;
+use Prophecy\Prophecy\ProphecySubjectInterface;
 use Psr\Container\ContainerInterface;
 use Psr\Http\Message\ResponseInterface;
 use ReflectionProperty;
 
 class BasicAccessFactoryTest extends TestCase
 {
-    /** @var ContainerInterface|ObjectProphecy */
+    /** @var ObjectProphecy */
```

Comment:
We should not drop the mocked type to avoid confusion.
What psalm error made you change these annotations?

Reply:
In the case of `ContainerInterface|ObjectProphecy` annotation Psalm will throw this errorr:

```
ERROR: PossiblyInvalidMethodCall - test\BasicAccessFactoryTest.php:75:15 - Cannot call method on possible bool variable  (see https://psalm.dev/113)
            ->willReturn(true);
```
The reason that there is not a common interface for these classes. and there is no `reveal` method in `ContainerInterface` as well as `willReturn` etc.

Reply:
@boesing this is an inherent problem of prophecy itself: the best approach here would be to replace prophecy with internal phpunit mocks, which come with proper typing too.

Up to @diwms if they want to do that or call this "done" here, because it is a bit more work.